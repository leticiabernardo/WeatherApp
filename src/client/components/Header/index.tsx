import { Box, Link } from '@chakra-ui/react';
import LanguageSwitcher from '@/client/components/LanguageSwitcher';
import TemperatureConverter from '@/client/components/TemperatureConverter';
import Search from '@/client/components/Search';

const Header = () => {
  return (
    <Box
      display="flex"
      alignItems="start"
      justifyContent="space-between"
      padding="10px 0 30px"
    >
      <Box margin="20px 0">
        <Link
          fontWeight="bold"
          color="white"
          fontSize="0.9em"
          href="https://weather.app"
          isExternal
        >
          weather.app
        </Link>
      </Box>

      <Box textAlign="right">
        <Box display="flex" justifyContent="space-around">
          <TemperatureConverter />
          <LanguageSwitcher />
        </Box>
        <Search />
      </Box>
    </Box>
  );
};

export default Header;
