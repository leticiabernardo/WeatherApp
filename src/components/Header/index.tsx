import { Box, Link } from '@chakra-ui/react';
import LanguageSwitcher from '@/components/LanguageSwitcher';
import TemperatureSwitcher from '@/components/TemperatureSwitcher';
import Search from '@/components/Search';

type HeaderProps = {
  setSearch: (val: string) => void;
  temperatureMeasurementUnit: app.MeasurementUnit;
  setTemperatureMeasurementUnit: (metric: app.MeasurementUnit) => void;
};

const Header = ({
  setSearch,
  temperatureMeasurementUnit,
  setTemperatureMeasurementUnit,
}: HeaderProps) => {
  return (
    <Box
      display="flex"
      alignItems="start"
      justifyContent="space-between"
      padding="10px 0 30px"
    >
      <Box margin="20px 0">
        <Link fontWeight="bold" color="white" fontSize="0.9em" href="./">
          weather.app
        </Link>
      </Box>

      <Box textAlign="right">
        <Box display="flex" justifyContent="space-around">
          <TemperatureSwitcher
            measurementUnit={temperatureMeasurementUnit}
            setMeasurementUnit={setTemperatureMeasurementUnit}
          />
          <LanguageSwitcher />
        </Box>
        <Search setSearch={setSearch} />
      </Box>
    </Box>
  );
};

export default Header;
