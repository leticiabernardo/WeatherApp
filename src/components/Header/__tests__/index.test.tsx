import { render, cleanup } from '@testing-library/react';
import Header from '../index';

const props = {
  setSearch: jest.fn(),
  setTemperatureMeasurementUnit: jest.fn(),
  setError: jest.fn(),
  temperatureMeasurementUnit: 'celsius' as app.MeasurementUnit,
};

describe('default wrapper render', () => {
  afterEach(cleanup);
  test('render the component', () => {
    const { container } = render(<Header {...props} />);
    expect(container).toMatchSnapshot();
  });
});
