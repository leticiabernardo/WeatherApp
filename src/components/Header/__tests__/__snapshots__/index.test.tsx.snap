// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default wrapper render render the component 1`] = `
<div>
  <div
    class="css-esbxfi"
  >
    <div
      class="css-1jlm3nb"
    >
      <a
        class="chakra-link css-3sqr2h"
        href="./"
      >
        weather.app
      </a>
    </div>
    <div
      class="css-s2uf1z"
    >
      <div
        class="css-1bnxa12"
      >
        <span
          class="css-10kghbw"
        >
          <button
            class="chakra-button css-yqy1ty"
            type="button"
          >
            <span
              class="css-0"
              tabindex="0"
            >
              °C
            </span>
          </button>
          |
          <button
            class="chakra-button css-hu7fh3"
            type="button"
          >
            <span
              class="css-0"
              tabindex="0"
            >
              °F
            </span>
          </button>
        </span>
        <span
          class="css-xe4d2t"
        >
          <button
            class="chakra-button css-hu7fh3"
            type="button"
          >
            EN
          </button>
          |
          <button
            class="chakra-button css-yqy1ty"
            type="button"
          >
            PT
          </button>
        </span>
      </div>
      <div>
        <div
          class="chakra-input__group css-4302v8"
        >
          <div
            class="chakra-input__left-element css-5tdink"
          >
            <button
              class="chakra-button css-dzjm3n"
              data-testid="geolocation-button"
              type="button"
            >
              <div
                class="svg-mock"
              />
            </button>
          </div>
          <input
            class="chakra-input css-ryoawo"
            placeholder="city, state or country..."
            value=""
          />
          <div
            class="chakra-input__right-element css-jwi0po"
          >
            <button
              class="chakra-button css-dzjm3n"
              data-testid="search-button"
              type="button"
            >
              <div
                class="svg-mock"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
